### PR TITLE
Added netmap and state update requirements

### PR DESCRIPTION
**Pull request by AtibQur**

### CODE REVIEW

File: sites/cheerpx/src/content/docs/11-guides/Networking.md [hunk 1]
```diff
@@ -66,8 +74,6 @@ Example to prompt the user for manual login on a different tab:
 ```
 
 ```js
-const loginElem = document.getElementById("loginLink");
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/11-guides/Networking.md:69]: This change is not intended I think

**Review by alexp-sssup — APPROVED** [review]

**Review by GabrielaReyna — CHANGES_REQUESTED** [review]

**Review by GabrielaReyna — CHANGES_REQUESTED** [review]
Great. Could you also do the same for ´controlUrl´ and ´authKey´ (Link and content in reference)

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 2]
```diff
@@ -79,3 +79,173 @@ networkInterface?: NetworkInterface;
...
+
+2. The `authKey` is added to the `networkInterface` object, ensuring the application can access the authentication data whenever needed in the CheerpX environment.
+
+### `controlUrl`
+
+```ts
+controlUrl?: string;
+```
+
+The `controlUrl` is an optional string parameter representing a URL used to manage or monitor the connection. It is typically fetched from the URL parameters and can be utilized for dashboard or configuration purposes.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:114]: Same as before. Simplify this in general. The purpose of this option is to select the self-hosted server, all the rest is noise.

In general prefer explaining what is the purpose of options from the user point of view,. Nobody cares about "the control plain", people just want to self host somehow

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 3]
```diff
@@ -79,3 +79,173 @@ networkInterface?: NetworkInterface;
...
+loginUrlCb?: (url: string) => void;
+```
+
+The `loginUrlCb` is a callback function designed to handle login URLs during the authentication process. It plays a critical role in the connection flow by setting the application's state to indicate readiness for login and resolving the URL required for the login process.
+
+**Example of usage:**
+
+```js
+function loginUrlCb(url) {
+	connectionState.set("LOGINREADY");
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:148]: The use of WebVM/Svelte specific code is not a good idea, since people will most likely not have this setup. It's not strictly required so it can be removed.

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 4]
```diff
@@ -79,3 +79,173 @@ networkInterface?: NetworkInterface;
...
+	resolveLogin(url);
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:149]: This is another WebVM idiom, we do this to update the networking login button when the login URL is ready, but this is a detail. Here we should strive for minimal self-contained examples. Printing the URL to the console or using window.open are better solution. I strongly recommend writing down a complete example to make sure everything works as intended.

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 5]
```diff
@@ -79,3 +79,173 @@ networkInterface?: NetworkInterface;
...
+
+This callback function is invoked whenever the connection state changes, enabling you to react dynamically to transitions such as connecting, disconnecting, or successfully establishing a connection. The `state` parameter is an integer representing the current state.
+
+**Example of usage:**
+
+```js
+function stateUpdateCb(state) {
+	switch (state) {
+		case 6: // Running
+			connectionState.set("CONNECTED");
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:200]: Remove WebVM specific code

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 6]
```diff
@@ -79,3 +79,173 @@ networkInterface?: NetworkInterface;
...
+	networkData.currentIp = map.self.addresses[0];
+	var exitNodeFound = false;
+	for (var i = 0; i < map.peers.length; i++) {
+		if (map.peers[i].exitNode) {
+			exitNodeFound = true;
+			break;
+		}
+	}
+	if (exitNodeFound) {
+		exitNode.set(true);
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:238]: Remove WebVM specific code

**Review by alexp-sssup — CHANGES_REQUESTED** [review]

File: sites/cheerpx/src/content/docs/11-guides/Networking.md [hunk 7]
```diff
@@ -56,8 +62,10 @@ const cx = await CheerpX.Linux.create({
 
 What is happening here?
 
-- `controlUrl` is a string URL of the Tailscale control plane which verifies the user's identity. You only need to pass this option if you are [self-hosting Tailscale](/docs/guides/Networking#self-hosting-headscale).
-- `authKey` is string with an auth key to register new users/devices that are pre-authenticated. You can create an auth key [here](https://login.tailscale.com/admin/settings/keys).
+- [authKey]: A string containing an authentication key for registering pre-authenticated users or devices. You can generate one [here](https://login.tailscale.com/admin/settings/keys).
+- [controlUrl]: The Tailscale control plane URL for identity verification. This is only needed if you are [self-hosting Tailscale](/docs/guides/Networking#self-hosting-headscale).
+- [stateUpdateCb]: A required callback function that monitors and reports changes in network status.
+- [netmapUpdateCb]: A callback function that provides updates on the network map, enabling access to network mapping information.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/11-guides/Networking.md:68]:
```suggestion
- [netmapUpdateCb]: A required callback function that provides updates on the network map, enabling access to the list of devices for the network.
```

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 8]
```diff
@@ -78,4 +78,116 @@ networkInterface?: NetworkInterface;
 
 This option configures network settings, which allows CheerpX to communicate over networks. For more detailed information about how CheerpX handles networking, including the use of Tailscale and overcoming browser limitations, see the [Networking](/docs/guides/Networking) guide.
 
+### `authKey`
+
+```ts
+authKey?: string;
+```
+
+The `authKey` is an optional string used for authentication and should be passed directly in the `create` call.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:87]: Reword this to be more similar to the brief explanation in the networking guide above, it's more accurate

Comment (alexp-sssup) [reply sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:87]: Please remove the "passed directly into" part. It's both inaccurate (since it's passed actually inside an object) and redundant since this whole page documents the `create` API

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 9]
```diff
@@ -78,4 +78,116 @@ networkInterface?: NetworkInterface;
...
+
+In this example, the `authKey` is included in the `networkInterface` object to enable authentication for the CheerpX instance.
+
+### `controlUrl`
+
+```ts
+controlUrl?: string;
+```
+
+The `controlUrl` is an optional string used to specify the URL of a self-hosted server.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:106]:
```suggestion
The `controlUrl` is an optional string used to specify the URL of a self-hosted Headscale server.
```

Also link back to the relevant guide. The short description in the Networking guide it's also better here

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 10]
```diff
@@ -78,4 +78,116 @@ networkInterface?: NetworkInterface;
...
+
+Here, the `controlUrl` is used to point the CheerpX instance to a custom self-hosted server.
+
+### `loginUrlCb`
+
+```ts
+loginUrlCb?: (url: string) => void;
+```
+
+The `loginUrlCb` is a callback function that handles login URLs during the authentication process. It provides the URL needed to continue the login process, such as authenticating with Tailscale.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:125]:
```suggestion
The `loginUrlCb` is a callback function that handles login URLs during the authentication process. It receives the URL that should be visited to continue the login process. This is necessary when authenticating with Tailscale.
```

**Review by alexp-sssup — CHANGES_REQUESTED** [review]

File: sites/cheerpx/src/content/docs/11-guides/Networking.md [hunk 11]
```diff
@@ -56,8 +62,10 @@ const cx = await CheerpX.Linux.create({
 
 What is happening here?
 
-- `controlUrl` is a string URL of the Tailscale control plane which verifies the user's identity. You only need to pass this option if you are [self-hosting Tailscale](/docs/guides/Networking#self-hosting-headscale).
-- `authKey` is string with an auth key to register new users/devices that are pre-authenticated. You can create an auth key [here](https://login.tailscale.com/admin/settings/keys).
+- [authKey]: A string containing an authentication key for registering pre-authenticated users or devices. You can generate one [here](https://login.tailscale.com/admin/settings/keys).
+- [controlUrl]: The Tailscale control plane URL for identity verification. This is only needed if you are [self-hosting Tailscale](/docs/guides/Networking#self-hosting-headscale).
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/11-guides/Networking.md:66]: This is not accurate. The control plain also coordinates access to the network, it's not just for authentication. Reword this to be something along the lines of: "When [self-hosting Headscale] you need to provide the control plain URL of your Headscale server. By default the main Tailscale control plane is used

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 12]
```diff
@@ -78,4 +78,116 @@ networkInterface?: NetworkInterface;
...
+Example:
+
+```js
+const cx = await CheerpX.Linux.create({
+	mounts: mountPoints,
+	networkInterface: { authKey: "YOUR KEY" },
+});
+```
+
+In this example, the `authKey` is included in the `networkInterface` object to enable authentication for the CheerpX instance.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:98]:
```suggestion
In this example, the `authKey` is included in the `networkInterface` object to automatically authenticate to Tailscale.
```

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 13]
```diff
@@ -78,4 +78,116 @@ networkInterface?: NetworkInterface;
...
+Example:
+
+```js
+const cx = await CheerpX.Linux.create({
+	mounts: mountPoints,
+	networkInterface: { controlUrl: "YOUR URL" },
+});
+```
+
+Here, the `controlUrl` is used to point the CheerpX instance to a custom self-hosted server.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:117]: Consider removing these extra comment lines, they are mostly redundant. The example is enough.

Comment (alexp-sssup) [reply sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:117]: They can be left if there is actually something to add after the example code

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 14]
```diff
@@ -78,4 +78,116 @@ networkInterface?: NetworkInterface;
...
+});
+
+function loginUrlCb(url) {
+	console.log("Login URL is ready:", url);
+	// Open the login URL in a new tab or window
+	window.open(url, "_blank");
+}
+```
+
+The `loginUrlCb` provides the user with a login URL for authentication. It logs the URL and opens it in a new tab or window. You can also adapt it to display the URL within your app or handle it differently as needed.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:142]:
```suggestion
The `loginUrlCb` receives a login URL for authentication. It logs the URL and opens it in a new tab or window. You can also adapt it to display the URL within your app or handle it differently as needed.
```

File: sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md [hunk 15]
```diff
@@ -78,4 +78,116 @@ networkInterface?: NetworkInterface;
...
+
+This example checks if the `state` indicates a "connected" status (`state = 6`) and logs "Connected" when the connection is established.
+
+### netmapUpdateCb
+
+```ts
+netmapUpdateCb?: (map: any) => void;
+```
+
+The `netmapUpdateCb` is a callback triggered whenever the network configuration updates. It provides details about the current network configuration.
```
Comment (alexp-sssup) [inline sites/cheerpx/src/content/docs/12-reference/00-CheerpX-Linux-create.md:175]:
```suggestion
The `netmapUpdateCb` is a callback that runs whenever the network configuration updates. It provides details about the current network configuration.
```

Be consistent with the previous one

**Review by alexp-sssup — CHANGES_REQUESTED** [review]

**Review by alexp-sssup — APPROVED** [review]

**Comment by alexp-sssup** [comment]
@GabrielaReyna LGTM
